app/controllers/api/account.php

```diff
@@ -1602,17 +1602,22 @@
 
         if (
             !$isAnonymousUser &&
+            !empty($user->getAttribute('password')) &&
```

Comment:
Unfortunately, this won't work because we actually set a random password for users if they sign in using OAuth2.